src/EventStore.Core/TransactionLog/Scavenging/DbAccess/ChunkReaderForIndexExecutor.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using EventStore.Core.TransactionLog.LogRecords;
+
+namespace EventStore.Core.TransactionLog.Scavenging {
+	public class ChunkReaderForIndexExecutor : IChunkReaderForIndexExecutor<string> {
+		private readonly Func<TFReaderLease> _tfReaderFactory;
+
+		public ChunkReaderForIndexExecutor(Func<TFReaderLease> tfReaderFactory) {
+			_tfReaderFactory = tfReaderFactory;
+		}
+
+		public bool TryGetStreamId(long position, out string streamId) {
+			using (var reader = _tfReaderFactory()) {
+				var result = reader.TryReadAt(position);
+				if (!result.Success ||
+					!(result.LogRecord is PrepareLogRecord prepare)) {
```

Comment:
should it be an exception if it's a non-prepare record?

Reply:
done

Reply:
thanks!